source/scripting_v3/GTA/Camera/CinematicCameraDirector.cs

```diff
@@ -0,0 +1,127 @@
+
+		/// <summary>
+		/// Invalidates the vehicle cinematic idle camera, restarting the associated idle counter.
+		/// </summary>
+		public static void InvalidateVehicleIdleCam() => Function.Call(Hash.INVALIDATE_CINEMATIC_VEHICLE_IDLE_MODE);
+
+		/// <summary>
+		/// Gets a value indicating whether a first person vehicle interior camera is active.
+		/// </summary>
+		/// <value>
+		/// <see langword="true" /> if the <see cref="CinematicCameraDirector"/> is a first person vehicle camera; otherwise, <see langword="false" />.
+		/// </value>
+		public static bool IsFirstPersonVehicleInteriorCamRendering => Function.Call<bool>(Hash.IS_CINEMATIC_FIRST_PERSON_VEHICLE_INTERIOR_CAM_RENDERING);
+
+		/// <summary>
+		/// Gets a value indicating whether the vehicle bonnet cam is rendering.
+		/// For example, this will return true if using the weapon cam on the Hydra.
+		/// Only available in v1.0.372.2 or later versions.
+		/// </summary>
+		/// <value>
+		/// <see langword="true" /> if the <see cref="CinematicCameraDirector"/> is a vehicle bonnet camera; otherwise, <see langword="false" />.
+		/// </value>
+		/// <exception cref="GameVersionNotSupportedException">
+		/// Thrown if called in game versions earlier than v1.0.372.2.
+		/// </exception>
+		public static bool IsVehicleBonnetCamRendering
+		{
+			get
+			{
+				GameVersionNotSupportedException.ThrowIfNotSupported(GameVersion.v1_0_372_2_Steam, nameof(CinematicCameraDirector), nameof(IsVehicleBonnetCamRendering));
+
+				return Function.Call<bool>(Hash.IS_BONNET_CINEMATIC_CAM_RENDERING);
+			}
+		}
+
+		/// <summary>
+		/// Gets a value indicating whether any cinematic camera is rendering.
+		/// Note that this will also return true if in first-person view while inside a vehicle.
+		/// </summary>
+		/// <value>
+		/// <see langword="true" /> if the <see cref="CinematicCameraDirector"/> is a cinematic camera; otherwise, <see langword="false" />.
+		/// </value>
+		public static bool IsAnyCamRendering => Function.Call<bool>(Hash.IS_CINEMATIC_CAM_RENDERING);
+
+		/// <summary>
+		/// Gets a value indicating whether an idle cinematic camera is rendering.
+		/// </summary>
+		/// <value>
+		/// <see langword="true" /> if the <see cref="CinematicCameraDirector"/> is an idle cinematic camera; otherwise, <see langword="false" />.
+		/// </value>
+		public static bool IsIdleCamRendering => Function.Call<bool>(Hash.IS_CINEMATIC_IDLE_CAM_RENDERING);
+
+		/// <summary>
+		/// Gets a value indicating whether a player-controlled cinematic camera is active.
+		/// Only available in v1.0.1493.0 or later game versions.
+		/// </summary>
+		/// <value>
+		/// <see langword="true" /> if the <see cref="CinematicCameraDirector"/> is a player-controlled cinematic camera; otherwise, <see langword="false" />.
+		/// </value>
+		/// <exception cref="GameVersionNotSupportedException">
+		/// Thrown if called in game versions earlier than v1.0.1493.0.
+		/// </exception>
+		public static bool IsPlayerCamActive
```

Comment:
How about `IsCinematicModeActive`? Since you can use cinematic cams other than the classic cinematic camera/mode such as the first person vehicle interior cam by pressing a certain button/key, `IsPlayerCamActive` may be too vague. It would be better we have better ideas than `IsCinematicModeActive` by distinct names though. Btw, `IS_CINEMATIC_CAM_INPUT_ACTIVE` returns true if the 4th bit at a certain offset is set,which can be set by the cinematic cam button or `SET_CINEMATIC_MODE_ACTIVE`, or the 8th bit at the next byte is set, which can be set by `FORCE_CINEMATIC_RENDERING_THIS_UPDATE`.

Here's the docs of `FORCE_CINEMATIC_RENDERING_THIS_UPDATE` and `SET_CINEMATIC_MODE_ACTIVE` in `commands_camera.sch`.
```
//INFO: If there is a cinematic camera rendering this cannot be overridden by the player. 
//PARAM NOTES:
//PURPOSE: Forces the cinematic mode to render this frame only. Will only render if there is a valid cinematic context. 
NATIVE PROC FORCE_CINEMATIC_RENDERING_THIS_UPDATE(BOOL ActiveThisFrame) = "0x683caf9e8ab4583d"
```
```
//INFO: If there is a cinematic camera rendering, this can be overriden by the player using quick toggle or pressing select
//PARAM NOTES:
//PURPOSE: Sets the cinematic mode to active. Will only render if there is a valid cinematic context else will be reset. 
NATIVE PROC SET_CINEMATIC_MODE_ACTIVE(bool Active) = "0x71800f6ecfd14497"
```